processing/src/main/java/org/apache/druid/query/lookup/LookupExtractorFactoryContainerProvider.java

```diff
@@ -40,4 +40,9 @@ public interface LookupExtractorFactoryContainerProvider
    * Returns a lookup container for the provided lookupName, if it exists.
    */
   Optional<LookupExtractorFactoryContainer> get(String lookupName);
+
+  /**
+   * Returns the canonical lookup name from a lookup name.
+   */
+  String getCanonicalLookupName(String lookupName);
```

Comment:
naming nit:
1. This can be getLookupName(String name). Canonical is confusing, and isn't defined in the javadocs. 

other comments:
1. This method isn't marked as `PublicApi` however custom extensions can extend it for own lookup implementations. Please add a default implementation for the same. 
2. When should this method be called. As of now, this is a super specialized method that is called in `QueryLookupOperatorConversion`. Perhaps it should be called out and named as such. As a developer, it is unclear to me when must I call it and when not. For example, why am I not calling it on https://github.com/apache/druid/blob/master/sql/src/main/java/org/apache/druid/sql/calcite/schema/LookupSchema.java#L61?

Reply:
@LakshSingla 
I had a discussion with @abhishekrb19 about the naming of this method. We concluded on getCanonicalLookupName().

Regarding the other comments:
1. I don't see the other methods marked as PublicApi either? So seems like an unrelated issue to this PR? Thoughts?
2. This would be used in Imply extension. Would raise an Imply PR once this merges where it would be more clear. But in general, if in future we add a lookup name with special syntaxing, then this would be needed.

Reply:
@LakshSingla, good points. I think we need to clarify the Javadoc a bit more to clarify the intent of the new method and/or rename accordingly.  We can also call it `getLookupName(String name)` too, but that's a bit ambiguous imo. Naming, thoughts? 😅 

As far as 1 is concerned, the interface isn't annotated `@PublicAPI` or `@ExtensionsPoint`. Also, this PR https://github.com/apache/druid/pull/9281/ changed the method signature and added a new interface method without default implementations. Following that pattern, I believe it's safe to add new methods without providing a default implementation. In general, my understanding is that for custom extensions that directly use/extend interfaces that are not public APIs (i.e., not annotated with `PublicAPI` or `ExtensionsPoint`), the onus is be on the developers maintaining custom implementations.

For 2, `LookupSchema` is powering the SQL view for the lookups configured in the system. I think it'd make sense to also call the new function there as you pointed out. I'm okay with doing that as a follow up too.